shared/src/main/scala/NumberHelpers.scala

```diff
@@ -103,6 +104,58 @@ object NumberHelpers:
         VList(result.toList*)
       case _ => throw new Exception("Cannot convert to binary")
 
+  def toBase(a: VAny, b: VAny)(using ctx: Context): VAny =
+    (a, b) match
+      case (a: VNum, b: VNum) =>
+        if b.toBigInt == 0 then 0
+        else toBaseDigits(a, b)
+      case (n: VNum, _)  => toBaseAlphabet(n, ListHelpers.makeIterable(b))
+      case (a: VList, _) => VList(a.map(toBase(_, b))*)
+      case _ => throw new Exception("toBase only works on numbers and lists")
+
+  /** Returns value in base len(alphabet) using base 10 [bijective base] */
+  def toBaseAlphabet(value: VNum, alphabet: VIter)(using ctx: Context): VAny =
+    val indexes = toBaseDigits(value, alphabet.iterLength)
+    val alphalist = VList((alphabet match
+      case a: String => a.toString.toList.map(_.toString())
+      case l: VList  => l
+    )*)
+
+    val temp = indexes.map(alphalist.index(_).toString())
+    alphabet match
+      case a: String => temp.mkString("")
+      case l: VList  => l
+
+  def toBaseDigits(value: VNum, base: VNum): VList =
+    if value == VNum(0) then VList(List(VNum(0))*)
+    else if base.toBigInt == -1 then
+      val out = ListBuffer[VNum]()
+      if value.toBigInt > 0 then
+        out += 1
+        for _ <- 1 until value.toInt do
+          out += 0
+          out += 1
+      else
+        val out = ListBuffer[VNum]()
+        for _ <- 1 until value.toInt.abs do
+          out += 1
+          out += 0
+      VList(out.toList*)
+    else
+      val sign = if value.toBigInt < 0 && base.toBigInt > 0 then -1 else 1
+      var current = sign * value.toBigInt
+      if base.toBigInt == 1 then VList(List.fill(current.toInt.abs)(sign)*)
+      else
+        val digits = ListBuffer[VAny]()
+        while current > 0 do
+          var digit = current % base.toBigInt
+          current = current / base.toBigInt
+          if digit < 0 then
+            current += 1
+            digit = digit - base.toBigInt
+          digits += VNum(digit * sign)
+        VList(digits.reverse.toList*)
```

Comment:
```suggestion
          VNum(digit * sign) +=: digits
        VList(digits.toList*)
```

Reply:
If we're prepending, we might as well do `var digits = List.empty[VNum]` and then do `digits = VNum(digit * sign) :: digits`, so at the end we can do `VList.from(digits)`

Reply:
Is that as efficient as using a mutable list?

Reply:
Okay, I did a bad benchmark and using a List is faster than a MutableList
https://scastie.scala-lang.org/U3UMcxCyTEGBaQBCin4xRg

Reply:
I figured it'd be more efficient since you don't have to create a new list at the end, it just gets built up normally (prepending is an O(1) operation on Lists). Efficiency doesn't matter too much here though, imo

Reply:
Hmm, I'm consistently getting the second ListBuffer approach to be faster than the List approach. Perhaps we should stick to ListBuffer then lol